src/frontend-samples/query-volume-app/VolumeQueryApp.tsx

```diff
@@ -0,0 +1,179 @@
+    ViewClipTool.enableClipVolume(vp);
+    // Turning off the clipping feature.
+    vp.view.viewFlags.clipVolume = isClippingOn === undefined ? false : isClippingOn;
+    vp.view.setViewClip(clip);
+    VolumeQueryApp.addDecorators(vp);
+  }
+
+  /* Method for adding decorators to the viewport */
+  public static addDecorators(vp: ScreenViewport) {
+    // Create a clip decorator. Selecting the clip decoration to immediately show the handles is the default.
+    const vcdp: ViewClipDecorationProvider = ViewClipDecorationProvider.create();
+    // Default behavior is to hide the decorators on deselect. We want to keep the decorators showing in this example.
+    vcdp.clearDecorationOnDeselect = false;
+    vcdp.showDecoration(vp);
+    // The decorators require the SelectTool being active.
+    IModelApp.toolAdmin.startDefaultTool();
+  }
+
+  /* Getting the range of currect box */
+  public static getRangeOfBox(vp: ScreenViewport): Range3d | undefined {
+    const viewClip = vp.view.getViewClip();
+    if (viewClip !== undefined) {
+      return ClipUtilities.rangeOfClipperIntersectionWithRange(
+        viewClip,
+        vp.view.computeFitRange(),
+      );
+    }
+
+    return undefined;
+  }
+
+  /* Getting all elements that are spatially represented in iModel*/
+  public static async getAllSpacialElements(vp: ScreenViewport): Promise<PhysicalElement[]> {
+    const esqlQuery = "SELECT ECInstanceId, ECClassId FROM BisCore.PhysicalElement";
+    const elementsAsync = vp.iModel.query(esqlQuery);
+    const elements: PhysicalElement[] = [];
+    for await (const element of elementsAsync) {
+      elements.push({ id: element.id, className: element.className, name: "" });
+    }
+
+    const presentationProvider = new PresentationLabelsProvider({ imodel: vp.iModel });
+    const packedInstancesKeys: InstanceKey[][] = [];
+    const instanceKeys: InstanceKey[] = elements.map((element) => {
+      return { className: element.className.replace(".", ":"), id: element.id };
+    });
+
+    /* Sorting elements into packages. Instead of one 3000 length array I will have 3 x 1000 lenght arrays */
+    const packsOfInstanceKeys = Math.floor(instanceKeys.length / 1000);
+    for (let i = 0; i <= packsOfInstanceKeys; i++) {
+      if (i !== packsOfInstanceKeys) {
+        packedInstancesKeys.push(instanceKeys.slice(i * 1000, (i + 1) * 1000));
+      } else {
+        packedInstancesKeys.push(instanceKeys.slice(i * 1000, instanceKeys.length + 1));
+      }
+    }
+
+    /* Getting elements names */
+    const elementNames: string[] = [];
+    for (const keysArray of packedInstancesKeys) {
+      const names = await presentationProvider.getLabels(keysArray);
+      elementNames.push(...names);
+    }
+
+    for (let i = 0; i < elementNames.length; i++)
+      elements[i].name = elementNames[i];
+
+    return elements;
+  }
+
+  /* Clear color from colored elements */
+  public static clearColoredSelection(vp: ScreenViewport) {
```

Comment:
Please rename to clearColorOverrides.  This really has nothing to do with Selection.